service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/CaseEventHandlerTest.java

```diff
@@ -0,0 +1,104 @@
+package uk.gov.hmcts.reform.fpl.handlers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+import uk.gov.hmcts.reform.fpl.events.CaseDataChanged;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.tasklist.Task;
+import uk.gov.hmcts.reform.fpl.service.TaskListRenderer;
+import uk.gov.hmcts.reform.fpl.service.TaskListService;
+import uk.gov.hmcts.reform.fpl.service.ccd.CoreCaseDataService;
+
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.commons.lang3.RandomUtils.nextLong;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.CASE_TYPE;
+import static uk.gov.hmcts.reform.fpl.CaseDefinitionConstants.JURISDICTION;
+import static uk.gov.hmcts.reform.fpl.enums.Event.CASE_NAME;
+import static uk.gov.hmcts.reform.fpl.enums.Event.SUBMIT_APPLICATION;
+import static uk.gov.hmcts.reform.fpl.enums.State.OPEN;
+import static uk.gov.hmcts.reform.fpl.enums.State.SUBMITTED;
+import static uk.gov.hmcts.reform.fpl.model.tasklist.TaskState.COMPLETED;
+import static uk.gov.hmcts.reform.fpl.model.tasklist.TaskState.NOT_AVAILABLE;
+
+@ExtendWith(SpringExtension.class)
```

Comment:
If you are mocking everything do we need to extend with spring? Could we not extend the mockito extension instead? Also are we sure we want to mock object mapper?

Reply:
of course, done